src/coreclr/src/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1199,6 +1251,10 @@ internal static RuntimeType ResolveTypeHandleInternal(RuntimeModule module, int
                 throw new ArgumentOutOfRangeException(nameof(typeToken),
                     SR.Format(SR.Argument_InvalidToken, typeToken, new ModuleHandle(module)));
 
+            // defensive copy of user-provided array, per CopyRuntimeTypeHandles contract
+            typeInstantiationContext = (RuntimeTypeHandle[]?)typeInstantiationContext?.Clone();
+            methodInstantiationContext = (RuntimeTypeHandle[]?)methodInstantiationContext?.Clone();
```

Comment:
Was this a potential GC bug in the original? i.e. no clone of `methodInstantiationContext`?

Reply:
Yes. The public entry point is `ModuleHandle.ResolveMethodHandle`, where the caller provides two `RuntimeTypeHandle[]` arguments. The possibility exists that the user invokes `ResolveMethodHandle`, we get the relevant pointers, then the thread is interrupted. Meanwhile, a second thread mutates the contents of those arrays. The first thread then resumes with the original pointers, but the `KeepAlive` at the end of the method is now protecting the _new_ handles instead of the _original_ handles which provided the pointers. If a GC kicks in here and reclaims any of the `RuntimeType` instances, it could destabilize the runtime.